napari/settings/_appearance.py

```diff
@@ -60,22 +76,36 @@ def update(
             values = values.dict()
         values = cast(dict, values)
 
-        # Check if a font_size change is needed when changing theme:
-        # If the font_size setting doesn't correspond to the default value
+        # Check if a `font_size` or `highlight.highlight_color` change is needed when changing theme:
+        # If the `font_size` or `highlight.highlight_color` setting doesn't correspond to the default value
         # of the current theme no change is done, otherwise
-        # the font_size value is set to the new selected theme font size value
+        # the `font_size` or `highlight.highlight_color` value is set to the new selected theme corresponding value
         if 'theme' in values and values['theme'] != self.theme:
             current_theme = get_theme(self.theme)
             new_theme = get_theme(values['theme'])
+            current_highlight_color = parse_color_as_float_list(
+                current_theme.border_highlight
+            ) + [1.0]
+            new_highlight_color = parse_color_as_float_list(
+                new_theme.border_highlight
+            ) + [1.0]
+
             if values['font_size'] == int(current_theme.font_size[:-2]):
                 values['font_size'] = int(new_theme.font_size[:-2])
+            if np.array_equal(
```

Comment:
```suggestion
            if np.allclose(
```
I think float precision is why changing theme doesn't change the widget color.